R/verb-expand.R

```diff
@@ -64,6 +54,23 @@ expand.tbl_lazy <- function(data, ..., .name_repair = "check_unique") {
   purrr::reduce(distinct_tables, left_join, by = group_vars(data))
 }
 
+extract_dot_vars <- function(dot) {
+  # ugly hack to deal with `nesting()`
+  if (quo_is_call(dot, name = "nesting")) {
```

Comment:
This hasn't worked before, but as we're touching this code:

Would that work if the user calls `tidyr::nesting()` in `expand()` ?

Also, do we want to implement an early return instead?

```suggestion
  if (!quo_is_call(dot, name = "nesting")) {
    return(...)
```

Reply:
Yes, it does work with `tidyr::nesting()`. By default `quo_is_call()` ignores the namespace. But it can be made more rigorous as follows

``` r
library(rlang)

ns <- c("tidyr", "")
quo_is_call(quo(nesting()), name = "nesting", ns = ns)
#> [1] TRUE
quo_is_call(quo(tidyr::nesting()), name = "nesting", ns = ns)
#> [1] TRUE
quo_is_call(quo(base::nesting()), name = "nesting", ns = ns)
#> [1] FALSE
```

<sup>Created on 2021-08-13 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>